gematria/testing/python/basic_blocks_with_throughput.py

```diff
@@ -155,7 +156,9 @@ class TestCase(unittest.TestCase):
   def setUp(self):
     super().setUp()
 
-    self.block_protos = get_basic_blocks(self.num_blocks)
+    self.block_protos = get_basic_blocks(
+        self.num_blocks, keep_fn=self.keep_function
+    )
```

Comment:
We should also add a test that `len(self.block_protos) == self.num_blocks` to be sure that we're not testing on an empty set and that we get a nice error message when the number of blocks is smaller than expected.

Reply:
Added a check that raises an error if we don't get the correct number of blocks.

Reply:
Ah, yep. Missed that the class was a test case for some reason.